packages/ui/src/AppSidebar/AppSidebarSection/AppSidebarSectionExpandable.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactNode, useContext, useEffect } from 'react'
+import cn from 'classnames'
+import { ChevronDown, ChevronUp, Icon as FeatherIcon } from 'react-feather'
+import createPersistedState from 'use-persisted-state'
+
+import { useRefValue } from '../../hooks'
+import { Icon, IconProps } from '../../Icon'
+import { AppSidebarItem } from '../AppSidebarItem'
+import { appSidebarContext } from '../appSidebarContext'
+import { appSidebarSectionContext } from './appSidebarSectionContext'
+
+import styles from './AppSidebarSection.module.scss'
+
+export interface AppSidebarSectionExpandableProps {
+  id: string
+  title: string
+  active: boolean
+  icon: FeatherIcon
+  ariaLabel: string
+  children: ReactNode
+}
+
+const kebabCase = (str: string) => str.toLowerCase().split(' ').join('-')
+
+export const AppSidebarSectionExpandable = ({ id, active, icon, title, ariaLabel, children }: AppSidebarSectionExpandableProps) => {
+  const { isOpen, toggle } = useContext(appSidebarContext)
+  const usePersistedMenuStorageState = createPersistedState<boolean>(`isOpen::${title}`)
+
+  const [open, setOpen] = usePersistedMenuStorageState(false)
+  const getIsActive = useRefValue(active)
+  const getSetOpen = useRefValue(setOpen)
+
+  const chevronButtonProps: IconProps = open
+    ? { icon: ChevronUp, ariaLabel: 'Section Open' }
+    : { icon: ChevronDown, ariaLabel: 'Section Closed' }
+  const isSectionOpen = open && isOpen
+
+  const handleClick = () => {
```

Comment:
Does this click handler and below use effect is for updating open/close state of a section?
Why do we have two different states here?
Can't we handle both states in single callback?

Reply:
useEffect - expands section if it has active item and sidebar is not collapsed
handleClick - expands sidebar or section